### PR TITLE
readonly volume should be bind mounted readonly on the host

### PR DESCRIPTION
**Pull request by bergwolf**
So that even if the guest is compromised, we still ensure that readonly volumes cannot be modified.

backport kata-containers/kata-containers#1062

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @bergwolf

**Comment by awprice** [comment]
Thanks for raising this @bergwolf - code works in my environment on 1.11.x and fixes the issue.

File: virtcontainers/mount.go [hunk 1]
```diff
@@ -273,6 +273,11 @@ func remount(ctx context.Context, mountflags uintptr, src string) error {
 	return nil
 }
 
+// remount a mount point as readonly
+func remount_ro(ctx context.Context, src string) error {
```
Comment (devimc) [inline virtcontainers/mount.go:277]: rust is affecting us :smile: 

```
virtcontainers/mount.go:277:6: don't use underscores in Go names; func remount_ro should be remountRo (golint)

func remount_ro(ctx context.Context, src string) error {
```

**Review by devimc — APPROVED** [review]
thanks @bergwolf - just one nit

**Comment by amshinde** [comment]
@chavafg Is the `jenkins-metrics-ubuntu-18-04` CI failing?
I don't see this PR causing a failure in the metrics. Can you take a look.

**Comment by chavafg** [comment]
The job looks stable, only failing in this PR with blogbench metric going out over the limit with a percentage value of 110.7%.

**Comment by amshinde** [comment]
I dont see how making volumes as read-only could impact the metrics. I'll rerun the the metrics CI to double check.

**Pull request merged**